wgengine/magicsock/magicsock.go

```diff
@@ -466,6 +470,7 @@ func NewConn(opts Options) (*Conn, error) {
 	c.port.Store(uint32(opts.Port))
 	c.controlKnobs = opts.ControlKnobs
 	c.logf = opts.logf()
+	c.cloudInfo.logf = c.logf
```

Comment:
do you still need this?

Reply:
Unfortunately yes, because we don't have `opts.logf` (and thus` c.logf`) in the `newConn` function.

Reply:
Refactored to have `newConn` take a `logf`, and cleaned things up.